### PR TITLE
Do not get internal interfaces when getting all interfaces

### PR DESCRIPTION
**Pull request by mattiaswal**
This is a regression from 2024.08

## Checklist

Tick *relevant* boxes, this PR is-a or has-a:

- [ ] Bugfix
  - [ ] Regression tests
  - [ ] ChangeLog updates (for next release)
- [ ] Feature
  - [ ] YANG model change => revision updated?
  - [ ] Regression tests added?
  - [ ] ChangeLog updates (for next release)
  - [ ] Documentation added?
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (please detail in commit messages)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

### CODE REVIEW

**Comment by troglobit** [comment]
Unfortunately this will not make it for the v24.09 release.

File: src/statd/python/yanger/yanger.py [hunk 1]
```diff
@@ -156,6 +156,13 @@ def iface_is_dsa(iface_in):
         return False
     return True
 
+def iface_is_internal(iface_in):
+    if "group" not in iface_in:
+        return False
+
+    if iface_in["group"] != "internal":
+        return False
+    return True
```
Comment (wkz) [inline src/statd/python/yanger/yanger.py:165]:
```suggestion
    return iface_in.get("group") == "internal"
```
`.get()` on dictionaries is pretty handy in these situations.

File: src/statd/python/yanger/yanger.py [hunk 2]
```diff
@@ -587,7 +594,6 @@ def get_brport_multicast(ifname):
 
     return multicast
 
-
```
Comment (wkz) [inline src/statd/python/yanger/yanger.py:590]: Isn't double-newlines some PEP8 thing?

**Review by wkz — APPROVED** [review]
Just some stylistic comments. Do with them what you will :smile:

**Comment by rical** [comment]
Right, I missed that. I will have a look during the day.

File: src/statd/python/yanger/yanger.py (hunk 2 above)
Comment (troglobit) [reply src/statd/python/yanger/yanger.py:590]: Indeed it is, between methods/classes/etc. outside of a class, iirc.

**Comment by rical** [comment]
I was already thinking about redesigning how Yanger fetches and iterates ip interface data. With this redesign this became even easier to solve. It also allowed me to cleanup the unit tests and add a test for the hiding of internal interfaces.

Have a look, it's in PR: #674 (supersedes this PR)

**Comment by mattiaswal** [comment]
Obsoleted by #674

**Pull request closed without merging**